Sources/GRPCHTTP2Core/GRPCStreamStateMachine.swift

```diff
@@ -1612,3 +1701,20 @@ extension MethodDescriptor {
     return "/\(self.service)/\(self.method)"
   }
 }
+
+extension Status {
+  @available(macOS 13.0, iOS 16.0, watchOS 9.0, tvOS 16.0, *)
+  fileprivate init(_ reason: GRPCStreamStateMachine.UnexpectedInboundCloseReason) {
+    switch reason {
+    case .streamReset:
+      self = Status(
+        code: .unavailable,
+        message: "Stream unexpectedly closed: a RST_STREAM frame was received."
+      )
+    case .channelInactive:
+      self = Status(code: .unavailable, message: "Stream unexpectedly closed.")
+    case .errorThrown(let error):
+      self = Status(code: .unavailable, message: "Stream unexpectedly closed with error: \(error).")
```

Comment:
I don't think we should include the error here: it's an internal error to the server. In most cases the client won't care but it could also leak internal details about the server that it shouldn't.